projects/packages/publicize/src/rest-api/class-jetpack-social-controller.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * The Jetpack Social Controller class.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+namespace Automattic\Jetpack\Publicize\REST_API;
+
+use Automattic\Jetpack\Connection\Traits\WPCOM_REST_API_Proxy_Request;
+use Automattic\Jetpack\Publicize\Publicize_Utils as Utils;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/**
+ * Jetpack Social Controller class.
+ */
+class Jetpack_Social_Controller extends Base_Controller {
```

Comment:
`Jetpack_Social_*` is a very generic name. I think it's better to have specific classes/controllers for specific endpoints. It may be better to name this class as `Shares_Data_Controller` and likewise the endpoint.

Reply:
The endpoint is called jetpack-social on wpcom, do you want to change that too?

Reply:
Also it returns more than just share data:
```
{
    "is_share_limit_enabled": true,
    "to_be_publicized_count": 0,
    "share_limit": 30,
    "publicized_count": 63,
    "shared_posts_count": 17,
    "shares_remaining": 0,
    "computed_shares_remaining": 0,
    "is_enhanced_publishing_enabled": true,
    "is_social_image_generator_enabled": true,
    "show_advanced_plan_upgrade_nudge": false
}
```

Reply:
Yes, we no longer use that other data.



> so perhaps we create this as a new `Shares_Data_Controller`, and we can deprecate the old one on WPCOM?

Yeah, that is better.